longevity_test.py

```diff
@@ -127,24 +138,27 @@ def test_custom_time(self):
             # In case we would like to verify all keys were written successfully before we start other stress / nemesis
             prepare_verify_cmd = self.params.get('prepare_verify_cmd', default=None)
             if prepare_verify_cmd:
-                verify_queue.append(self.run_stress_thread(stress_cmd=prepare_verify_cmd, keyspace_num=keyspace_num))
+                self._run_all_stress_cmds(verify_queue, params={'stress_cmd': prepare_verify_cmd,
+                                                               'keyspace_num': keyspace_num})
 
                 for stress in verify_queue:
                     self.verify_stress_thread(queue=stress)
 
-        # Stress: Same as in prepare_write - allow the load to be spread across all loaders when using MULTI-KEYSPACES
-        if keyspace_num > 1 and self.params.get('round_robin', default='').lower() == 'true':
-                self.log.debug("Using round_robin for multiple Keyspaces...")
-                for i in xrange(1, keyspace_num + 1):
-                    keyspace_name = 'keyspace{}'.format(i)
-                    params = {'keyspace_name': keyspace_name, 'round_robin': True}
+        stress_cmd = self.params.get('stress_cmd', default=None)
+        if stress_cmd:
+            # Stress: Same as in prepare_write - allow the load to be spread across all loaders when using MULTI-KEYSPACES
+            if keyspace_num > 1 and self.params.get('round_robin', default='false').lower() == 'true':
+                    self.log.debug("Using round_robin for multiple Keyspaces...")
+                    for i in xrange(1, keyspace_num + 1):
+                        keyspace_name = self._get_keyspace_name(i)
+                        params = {'keyspace_name': keyspace_name, 'round_robin': True, 'stress_cmd': stress_cmd}
 
-                    self._run_all_stress_cmds(stress_queue, params)
+                        self._run_all_stress_cmds(stress_queue, params)
 
-        # The old method when we run all stress_cmds for all keyspace on the same loader
-        else:
-                params = {'keyspace_num': keyspace_num}
-                self._run_all_stress_cmds(stress_queue, params)
+            # The old method when we run all stress_cmds for all keyspace on the same loader
+            else:
+                    params = {'keyspace_num': keyspace_num, 'stress_cmd': stress_cmd}
+                    self._run_all_stress_cmds(stress_queue, params)
 
         # Check if we shall wait for total_used_space or if nemesis wasn't started
         if not prepare_write_cmd or self.params.get('nemesis_during_prepare', default='true').lower() == 'false':
```

Comment:
I think you can remove the sleep(600) below and use the fixed wait_for_total_space now.